atecc508a_config_dev.h

```diff
@@ -0,0 +1,182 @@
+ * Additionally, slot 6 has a different ECDH operation mode, to use slot 7 as
+ * a storage for ECDH master secret.
+ * Slot 8 is a certificate storage with clear read and write enabled.
+ * Slots 9-14 are configured as public keys with clear read and write enabled.
+ * Slot 15 is a private key that utilizes a limited use counter, unique to
+ * slot 15. */
+
+const uint8_t config_508a_dev[]  =
+{
+  /* 0-15 are read-only device-specific bytes, but for CRC calculation during
+   * config writing/locking these have to be read from the device. */
+  0x00, 0x00, 0x00, 0x00, /* 0-3 First part of serial number */
+  0x00, 0x00, 0x00, 0x00, /* 4-7 Device revision number */
+  0x00, 0x00, 0x00, 0x00, 0x00, /* 8-12 Second part of the serial number */
+  0x00, /* 13 Reserved */
+  0x00, /* 14 I2C Enable */
+  0x00, /* 15 Reserved */
+  /* End of skipped zone */
+
+  0xC0, /* 16 I2C Address, default value */
+  0x00, /* 17 Reserved */
+  0x55, /* 18 OTP Mode - consumption (removed in ATECC608A) */
+  0x00, /* 19 Chip Mode - Watchdog is 1.3s (as recommended), Selector always
+         * writable, fixed input voltage reference. */
+
+  /* Bytes 20-51 are SlotConfig, 2 bytes per slot. Slots 0-7: private keys.
+   * 0x8720 = 1000 0111 0010 0000
+   * (12-15) 0010   WriteConfig - Clear text write permitted, GenKey may be
+   *                              used to write random keys into this slot.
+   * (8-11)  0000   WriteKey - Irrelevant, since clear writes are permitted.
+   * (7)     1      IsSecret - The contents of this slot are secret.
+   * (6)     0      EncryptRead - Reads from this slot are prohibited.
+   * (5)     0      LimitedUse - Unlimited use.
+   * (4)     0      NoMac - The key stored in the slot can be used by
+   *                        all commands.
+   * (0-3)   0111   ReadKey - (priv key in slot) - External and internal
+   *                          signatures are enabled, ECDH operations too. */
+  0x87, 0x20, /* 20-21 SlotConfig 0 */
+  0x87, 0x20, /* 22-23 SlotConfig 1 */
+  0x87, 0x20, /* 24-25 SlotConfig 2 */
+  0x87, 0x20, /* 26-27 SlotConfig 3 */
+  0x87, 0x20, /* 28-29 SlotConfig 4 */
+  0x87, 0x20, /* 30-31 SlotConfig 5 */
+  /* Slot 6 has a different ECDH operation mode - instead of outputing
+   * ECDH master secret in the clear - it uses slot n+1 (7) to write it to.
+   * 0x8F20, ReadKey is 1111. */
+  0x8F, 0x20, /* 32-33 SlotConfig 6 */
+  0x87, 0x20, /* 34-35 SlotConfig 7 */
+
+  /* Slot 8: data storage, slots 9-14: public keys - 0x0000
+   * (12-15) 0000   WriteConfig - Clear text write permitted.
+   * (8-11)  0000   WriteKey - Irrelevant, since clear writes are permitted.
+   * (7)     0      IsSecret - Enabled clear text reads.
+   * (6)     0      EncryptRead - Enabled clear text reads.
+   * (5)     0      LimitedUse - Unlimited use.
+   * (4)     0      NoMac - The key stored in the slot can be used by all commands.
+   * (0-3)   0000   ReadKey - This slot can be the source for the CheckMac/Copy operation. */
```

Comment:
Can we verify signatures using public keys in these slots with `ReadKey` being `0000`?

Reply:
Yes, as the datasheet treats the data in this place differently depending on whether the key is private or public:
- Public keys:
`0x0 = Then this slot can be the source for the CheckMac/Copy operation. `
- Private keys:
`Slots containing private keys can never be read and this field has a different meaning:`
Followed by some specific rules regarding signatures (but only when using a private key in this slot).